lib/aganakti/client.rb

```diff
@@ -0,0 +1,118 @@
+# frozen_string_literal: true
+
+module Aganakti
+  ##
+  # An instance of a Druid SQL client. Because Typhoeus/libcurl handle all thread and process safety issues, it is not necessary to create a
+  # new client per thread/process (e.g., in a Unicorn after_fork). The client is intended to be constructed by {Aganakti.new}, which abstracts
+  # some of the more complicated cURL options. You're welcome to construct this class instead if you need to do something special.
+  #
+  # The main method on this class is {#query}, which allows you to construct an optionally parameterized query.
+  #
+  # This class additionally has some helpers for escaping when parameterized queries are unavailable or are unsupported in the
+  # situation you're in (e.g., +INTERVAL '5' MONTHS+ cannot use a parameter for +'5'+). These methods are on the client instance rather than
+  # the client class to provide for the possibility of server-specific escaping rules in the future.
+  class Client
+    ##
+    # @api private
+    # @return [ActiveSupport::Notifications::Instrumenter] the ActiveSupport::Notifications instrumenter
+    attr_reader :instrumenter
+
+    ##
+    # @return [Hash] The options that are passed to Typhoeus
+    attr_reader :typhoeus_options
+
+    ##
+    # @return [String] The URI to make queries against
+    attr_reader :uri
+
+    ##
+    # Creates a new client instance. You probably want to use {Aganakti.new} unless you need to specify your own Typhoeus options.
```

Comment:
We should probably mention here, that this overrides the `Accept` and `Content-Type` members within the `:headers` option.